io/net/net_tools.py

```diff
@@ -0,0 +1,374 @@
+            self.log.info("verified -%s option of hostname command" % option)
+
+        # Test to change hostname
+        process.system("hostname -F %s" % self.myhostname_file,
+                       verbose=False,
+                       sudo=True)
+        self.restore_hostname = True
+        output = process.run("hostname", verbose=False,
+                             env={"LANG": "C"})
+        if "myhost.my-domain" not in output.stdout:
+            self.error("unexpected response from hostname -F command and " +
+                       "hostname -F didn't set hostname")
+        ret = self._restore_hostname(self.hostname_file, self.hostname)
+        if ret:
+            self.error(ret)
+        self.restore_hostname = False
+
+    @staticmethod
+    def _restore_hostname(hostnamefile, hostname):
+        # Restore hostname
+        process.system("hostname -F %s" % hostnamefile,
+                       verbose=False)
+        output = process.run("hostname", verbose=False,
+                             env={"LANG": "C"})
+        if hostname not in output.stdout:
+            return "Failed to restore Hostname"
+
+    def tearDown(self):
+        if self.restore_hostname:
+            ret = self._restore_hostname(self.hostname_file, self.hostname)
+            if ret:
+                self.error(ret)
+        shutil.rmtree(self.backup_dir)
+
+
+class Ifconfig(Test):
+    """
+    ifconfig - configure a network interface
+    Test to verify the ifconfig functionality.
+    """
+
+    def setUp(self):
+        if "0% packet loss" not in process.system_output(
+                "ping -c 5 -w 5 localhost", verbose=False):
+            self.skip("Loopback interface is in down state")
+
+        self.lo_intf_state = "Up"
+        if "lo:1" in process.system_output("ifconfig", verbose=False):
+            self.skip("alias for an loopback interface is configured")
+        self.alias = None
+        install_dependencies()
+        self.ipv6 = False
+        if os.path.exists("/proc/net/if_inet6"):
+            self.ipv6 = True
+
+    def test_ifconfig(self):
+        """
+        Verify the functionality of the ifconfig
+        """
+        output = process.run("ifconfig", verbose=False,
+                             env={"LANG": "C"})
+        if "Local Loopback" not in output.stdout:
+            self.error("unexpected output of ifconfig")
+        if self.ipv6:
+            if "inet6" not in output.stdout:
+                self.error("Did not see IPV6 info")
+        # setup and verify an alias interface
+        self.alias = "lo:1"
+        process.system("ifconfig lo:1 127.0.0.240  netmask 255.0.0.0",
+                       verbose=False,
+                       sudo=True)
+        output = process.run("ifconfig", verbose=False,
+                             env={"LANG": "C"})
+        if "lo:1" not in output.stdout:
+            self.error("Failed to configure alias for loopback")
```

Comment:
^^  regarding the `verbose=False`, you can keep it disabled to shrink the log, but then you have to log the whole results in case of error (or failure)